static/src/javascripts/projects/common/modules/charts/doughnut.js

```diff
@@ -0,0 +1,185 @@
+// @flow
+/*
+ * Inspired by
+ * - http://www.chartjs.org/
+ * - http://codepen.io/githiro/pen/ICfFE
+ * - https://github.com/mbostock/d3/blob/master/src/svg/arc.js
+ */
+import $ from 'lib/$';
+import bonzo from 'bonzo';
+
+const svgEl = (type: string): bonzo =>
+    $.create(document.createElementNS('http://www.w3.org/2000/svg', type));
+
+const translate = (v: Array<number>): string => `translate(${v.toString()})`;
+
+/**
+ * @param {Object.<string, *>} data in the format { label: string, value: number, color: HEX }
+ * @param {Object.<string, *>} o the options
+ * @return {Bonzo} SVG Element
+ */
+const Doughnut = (data: Object, o: Object): bonzo => {
+    let obj: Object = o;
+    obj = Object.assign(
+        {
+            percentCutout: 35,
+            unit: '',
+            showValues: false,
+        },
+        obj || {}
+    );
+
+    const w = obj.width;
+    const h = obj.height || w;
+    const radius = Math.min(h / 2, w / 2);
+    const cutoutRadius = radius * (obj.percentCutout / 100);
+
+    const totalValue = data.reduce((a, b) => ({
+        value: a.value + b.value,
+    })).value;
+
+    const halfPI = Math.PI / 2;
+    const doublePI = Math.PI * 2;
+    const c = [w / 2, h / 2];
+
+    const center = {
+        x: w / 2,
+        y: h / 2,
+    };
+
+    const $svg = $.create(
+        '<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" class="chart chart--doughnut"></svg>'
+    ).attr({
+        width: w,
+        height: h,
+        viewbox: `0 0 ${[w, h].join(' ')}`,
+    });
+
+    // Segments
```

Comment:
Could these be declared within the `data.forEach` block? They aren't referenced outside of each `forEach` iteration.